### PR TITLE
Dashboard fix pod status

### PR DESCRIPTION
**Pull request by zehuaiWANG**
refer to issue #4740 

I found that  dashboard can not handle the pod terminating status,
BTW, when the pod phase is Running and the DeletionTimestamp is not nil, dashboard will show the pod is Running, but it's Terminating if I use kubectl describe pod

cc @maciaszczykm @floreks @shu-mutou 
Please don't merge it now. I think it should be discussed. Maybe the front-end should be changed either.

Above it's some reference about pod status:
[How to use the kubernetes go-client to get the same Pod status info that kubectl gives](https://stackoverflow.com/questions/57297817/how-to-use-the-kubernetes-go-client-to-get-the-same-pod-status-info-that-kubectl)
[kubernetes/kubernetes printpod](https://github.com/kubernetes/kubernetes/blob/master/pkg/printers/internalversion/printers.go#L558)

### CODE REVIEW

**Comment by floreks** [comment]
I'd leave base statuses as a list of:
- Running
- Pending
- Succeded
- Failed

We could display additional information on the details view, but on the list view I'd stick to the above 4.

**Comment by zehuaiWANG** [comment]
@floreks okay, I agree with you that display the base statuses as a list of: Running、Pending、Succeded、Failed. But I think we should add some code to handle the case if the pod is terminating because when the pod status is terminating, the pod is not use anymore, but in dashboard, we still display it as running.

**Comment by zehuaiWANG** [comment]
some code such like:
```
if pod.DeletionTimestamp != nil {
		return v1.PodFailed			reason = "Terminating"
	}		
}
```
we should take `pod.DeletionTimestamp` into consideration.

**Comment by floreks** [comment]
So maybe we could add another status like `Terminating`. What is the status of pod in kubectl when it is being deleted?

**Comment by zehuaiWANG** [comment]
when a pod is being deleted:

- The object is still visible via the REST API

- The object’s deletionTimestamp is set

And the pod status display in `kubectl get pod xxx` will be `Terminating`. But in general, the pod will be deleted quickly, which makes it difficult for us to see the `Termination` state. However, if the pod cannot be deleted for some reasons (for example, the resource of the pod is occupied by other processes), it will be in the `Termination` state.

More than that, I think we may need to redetermine the status of pods, because if a node loses contact, then all the pods on the node will be in the `Unknown` state. There are also some cases, such as image pull failure or container startup failure. Do we set the pod status to `Failed`? Because the `Failed` state in the kubectl command actually means that the pod error has exited and will not be restarted, and the container pull failure will be restarted.

This PR is written according to `kubectl get pod` command in kubernetes, but I think it needs to be discussed and decided, so I hold him.

@floreks

**Comment by zehuaiWANG** [comment]
crash
Terminating
such like this. @floreks

**Comment by floreks** [comment]
We can add `Terminating` as additional status. For containers that are constantly restarted due to some error we have `Pending` status + error icon and we can click on the pod to see related error. If we can determine that node has lost connection and Pods are in an `Unknown` state then I'd add that also to the list of states. We would present some icon with the question mark then as a status icon.

**Comment by zehuaiWANG** [comment]
okay, I agree with you. And I'd modify this pr and add `Unknow` and `Terminating`  to the list of states later. Thanks.

File: src/app/backend/resource/pod/common.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ func getPodStatus(pod v1.Pod, warnings []common.Event) PodStatus {
 	}
 }
 
-// getPodStatusPhase returns one of four pod status phases (Pending, Running, Succeeded, Failed)
+// getPodStatusPhase returns one of four pod status phases (Pending, Running, Succeeded, Failed, Unknow, Terminating)
```
Comment (floreks) [inline src/app/backend/resource/pod/common.go]: s/Unknow/Unknown

**Comment by maciaszczykm** [comment]
@zehuaiWANG Will you update it?

**Comment by zehuaiWANG** [comment]
Yes. This is an important PR. Sorry I forgot about it, I think I need to look at it again to make sure there are no new problems. Maybe about next week I will update it.

**Comment by zehuaiWANG** [comment]
@maciaszczykm I've already tested it and found no problem. Thank you for fixing the typo.

**Comment by maciaszczykm** [comment]
It's problem with Travis.

**Pull request merged**